frontend/src/components/assignment/create/AssignmentEditor.tsx

```diff
@@ -49,10 +50,17 @@ export default function AssignmentEditor({ isEditing = false }: Props) {
       setIsPublished(assignment.isPublished);
     }
   }, []);
-
-  const { user } = useUserContext();
-
   const { toast } = useToast();
+  const { user } = useUserContext();
+  if (!(user && Cookies.get('token'))) {
```

Comment:
I think it is not needed to check the existence of "token" since if they are not present, middleware will reroute to login page, can refer to [here](https://github.com/tryyang2001/CS3213-Frontend-Management-System/blob/master/frontend/src/middleware.ts#L18) in master branch.
![image](https://github.com/tryyang2001/CS3213-Frontend-Management-System/assets/69421979/28c481eb-9704-4360-8904-fa3737e79af8)

Reply:
I thought previously you said we won't have middleware for frontend anymore? but okay

Reply:
Also if we're having middleware instead of checking user context and token in components, then i think we need to use toast or whatever to show some kind of error message when you redirect them to `/login` since for now it will be confused for user without any messages. BecauseI was under the assumption that we don't have middleware in frontend while debugging since the last time we talked about this and it took me until now reading this comment to know why I cannot route to another page even after logging in and toast message log in successful show.

Reply:
I think that is okay though? When you tried to visit a page, and you are redirected to login page, you immediately know that you need to log in? Am I wrong? Having middleware is just to provide a centralised control for the redirection, but middleware itself is server side component, so we cannot and should not add toasting.